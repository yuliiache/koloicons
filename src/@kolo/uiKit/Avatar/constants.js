export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
