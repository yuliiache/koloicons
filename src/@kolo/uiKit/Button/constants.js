export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
