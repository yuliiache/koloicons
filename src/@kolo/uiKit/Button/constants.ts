export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
