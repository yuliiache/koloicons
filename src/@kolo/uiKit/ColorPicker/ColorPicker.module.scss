@import "@kolo/design";

.label {
  display: flex;
  justify-content: center;
  align-items: center;

  background-color: $IconInvert;
  border: 0;
  border-radius: 50%;
  box-shadow: $ColorPickerBoxShadow;
  cursor: pointer;

  &.big {
    width: 40px;
    height: 40px;

    & > .colorPointer {
      width: 20px;
      height: 20px;
    }
  }

  &.small {
    width: 32px;
    height: 32px;

    & > .colorPointer {
      width: 16px;
      height: 16px;
    }
  }
}

.colorPointer {
  border: 0;
  border-radius: 50%;
}

.colorInput {
  @include vh;
}
