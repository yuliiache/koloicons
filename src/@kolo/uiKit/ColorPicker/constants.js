export const COLORPICKER_TEST_ID = 'colorPickerComponent';

export const COLORPICKER_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
});

export const COLORPICKER_DEFAULT_COLOR = '#000000';
