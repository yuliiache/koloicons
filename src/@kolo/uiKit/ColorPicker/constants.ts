export const COLORPICKER_TEST_ID = 'colorPickerComponent';

export enum ColorPickerSize {
  SMALL = 'small',
  BIG = 'big',
}

export enum ColorPickerColor {
  DEFAULT = '#000000',
}
