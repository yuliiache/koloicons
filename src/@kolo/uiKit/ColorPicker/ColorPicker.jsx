import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import {COLORPICKER_DEFAULT_COLOR, COLORPICKER_SIZE, COLORPICKER_TEST_ID} from './constants';

import styles from './ColorPicker.module.scss';

const ColorPicker = ({onChange, size = COLORPICKER_SIZE.BIG, value = COLORPICKER_DEFAULT_COLOR}) => {
  const labelClassName = classNames(styles.label, styles[`${size}`]);

  return (
    <label className={labelClassName}>
      <div
        className={styles.colorPointer}
        style={{backgroundColor: value}}
      />
      <input
        className={styles.colorInput}
        type="color"
        id="colorPicker"
        value={value}
        onChange={onChange}
        data-testid={COLORPICKER_TEST_ID}
      />
    </label>
  );
};

ColorPicker.propTypes = {
  size: PropTypes.oneOf(Object.values(COLORPICKER_SIZE)),
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

export default ColorPicker;
