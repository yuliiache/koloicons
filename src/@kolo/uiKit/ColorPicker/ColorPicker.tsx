import classNames from 'classnames';
import {FC} from 'react';

import styles from './ColorPicker.module.scss';
import {COLORPICKER_TEST_ID, ColorPickerColor, ColorPickerSize} from './constants';

interface ColorPickerProps {
  onChange: () => void;
  size?: ColorPickerSize;
  value?: ColorPickerColor;
}

const ColorPicker: FC<ColorPickerProps> = ({
  onChange,
  size = ColorPickerSize.BIG,
  value = ColorPickerColor.DEFAULT,
}) => {
  const labelClassName = classNames(styles.label, styles[`${size}`]);

  return (
    <label className={labelClassName}>
      <div
        className={styles.colorPointer}
        style={{backgroundColor: value}}
      />
      <input
        className={styles.colorInput}
        type="color"
        id="colorPicker"
        value={value}
        onChange={onChange}
        data-testid={COLORPICKER_TEST_ID}
      />
    </label>
  );
};

export default ColorPicker;
