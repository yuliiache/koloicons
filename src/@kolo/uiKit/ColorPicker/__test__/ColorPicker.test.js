import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';

import ColorPicker from '../ColorPicker';
import {COLORPICKER_TEST_ID,ColorPickerSize} from '../constants';

describe('ColorPicker component', () => {
  Object.values(ColorPickerSize).forEach((size) => {
    const onChange = jest.fn();
    const props = {
      size,
      onChange,
      value: '#000000',
    };

    it(`should renders correctly with ${size} size`, () => {
      const {asFragment} = render(<ColorPicker {...props} />);

      expect(asFragment()).toMatchSnapshot();
    });

    it(`with ${size} size triggers onChange event`, () => {
      render(<ColorPicker {...props} />);

      const colorPicker = screen.getByTestId(COLORPICKER_TEST_ID);
      const newColor = '#269964';

      fireEvent.change(colorPicker, {target: {value: newColor}});

      expect(onChange).toHaveBeenCalledTimes(1);
    });
  });
});
