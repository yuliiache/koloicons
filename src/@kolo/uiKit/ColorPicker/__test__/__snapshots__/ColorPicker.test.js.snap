// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component should renders correctly with big size 1`] = `
<DocumentFragment>
  <label
    class="label big"
  >
    <div
      class="colorPointer"
      style="background-color: rgb(0, 0, 0);"
    />
    <input
      class="colorInput"
      data-testid="colorPickerComponent"
      id="colorPicker"
      type="color"
      value="#000000"
    />
  </label>
</DocumentFragment>
`;

exports[`ColorPicker component should renders correctly with small size 1`] = `
<DocumentFragment>
  <label
    class="label small"
  >
    <div
      class="colorPointer"
      style="background-color: rgb(0, 0, 0);"
    />
    <input
      class="colorInput"
      data-testid="colorPickerComponent"
      id="colorPicker"
      type="color"
      value="#000000"
    />
  </label>
</DocumentFragment>
`;
